### PR TITLE
chore: merge pegasus-scala-client to incubator-pegasus

### PR DESCRIPTION
**Pull request by cauchy1988**
### What problem does this PR solve? 
merge pegasus scala client into incubator-pegasus

details can be seen in the issue : https://github.com/apache/incubator-pegasus/issues/945

### What is changed and how does it work?
only add the scala client repo

##### Tests 

- Unit test
- Integration test

### CODE REVIEW

File: scala-client/README.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+# pegasus-scala-client
+
+This is the scala client for [apache/incubator-pegasus](https://github.com/apache/incubator-pegasus).
+
+It's built on top of the java
+client [apache/incubator-pegasus/java-client](https://github.com/apache/incubator-pegasus/tree/master/java-client).
```
Comment (acelyc111) [inline scala-client/README.md:24]:
```suggestion
It's built on top of the java client [apache/incubator-pegasus/java-client](https://github.com/apache/incubator-pegasus/tree/master/java-client).
```

File: scala-client/README.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+software distributed under the License is distributed on an
+"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+KIND, either express or implied.  See the License for the
+specific language governing permissions and limitations
+under the License.
+-->
+# pegasus-scala-client
+
+This is the scala client for [apache/incubator-pegasus](https://github.com/apache/incubator-pegasus).
+
```
Comment (acelyc111) [inline scala-client/README.md:22]:
```suggestion
```
Meaningless now, we can remove it.

**Comment by acelyc111** [comment]
I've checked:

- [ ] .github/workflows/*.yaml
- [ ] .gitignore
- [ ] com.xiaomi.infra.pegasus renamed to org.apache.pegasus
- [ ] new actions are running normally
- [ ] All files have APLv2 header
- [ ] SER[V] -> Serializer[V]

**Review by acelyc111 — APPROVED** [review]
Good job!

**Review by acelyc111 — APPROVED** [review]

**Review by neverchanje — APPROVED** [review]
lgtm

**Pull request merged**